src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,121 @@
+    private String path;
+    private Ui ui;
+    public Storage(String path, Ui ui) {
+        this.path = path;
+    }
+    public static LocalDateTime stringToDateTime(String str) throws DateTimeParseException {
+        //check if dateTime has correct format: ie. YYYY-MM-DD 00:00
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        LocalDateTime dateTime = LocalDateTime.parse(str, formatter);
+        return dateTime;
+    }
+    public File createDataFile() {
+        File dataFile = new File(this.path);
+        try {
+            Path dirPath = Paths.get("./data/");
+            if (!Files.exists(dirPath)) {
+                Files.createDirectory(dirPath);
+            }
+            if (!dataFile.exists()) {
+                dataFile.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println("Unable to create file!!");
+        }
+        return dataFile;
+    }
+
+    public ArrayList<Task> loadTasks() {
+        File dataFile = new File(this.path);
+        if (!dataFile.exists()) {
+            dataFile = createDataFile();
+        }
+        ArrayList<Task> list = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(dataFile);
+            while (sc.hasNextLine()) {
+                String task = sc.nextLine();
+                if (!task.isBlank()) {
+                    // | is a special symbol
+                    String[] taskDetails = task.split(" " + "\\|" + " ");
+                    String type = taskDetails[0];
+                    int status = Integer.parseInt(taskDetails[1]);
+                    String desc = taskDetails[2];
+                    switch (type) {
+                    case "T":
+                        ToDo toDo = new ToDo(status, desc);
+                        list.add(toDo);
+                        break;
+                    case "D":
+                        LocalDateTime date = stringToDateTime(taskDetails[3]);
+                        Deadline deadline = new Deadline(status, desc, date);
+                        list.add(deadline);
+                        break;
+                    case "E":
+                        LocalDateTime start = stringToDateTime(taskDetails[3]);
+                        LocalDateTime end = stringToDateTime(taskDetails[4]);
+                        try {
+                            Event event = new Event(status, desc, start, end);
+                            list.add(event);
+                        } catch (InvalidStartEndException e) {
+                            ui.showError(e.getMessage());
+                        }
+                        break;
+                    }
+                }
+            }
+            sc.close();
+        } catch (IOException e) {
+            ui.showError(e.getMessage());
+        }
+        return list;
+    }
+
+    //this method should be called by the tasklist class ONLY!!! because we want to
+    //keep the tasklist private
```

Comment:
I think rather than adding a comment, it would be better for TaskList to generate a ArrayList of strings for the Storage class to write to file. This ensures that the the ArrayList of Tasks would not be passed to an external class. This make the Storage class more reusable as well.